persistent/Database/Persist/TH.hs

```diff
@@ -473,6 +473,15 @@ mkEntityDefSqlTypeExp emEntities entityMap ent =
 -- 'EntityDef's. Works well with the persist quasi-quoter.
 mkPersist :: MkPersistSettings -> [EntityDef] -> Q [Dec]
 mkPersist mps ents' = do
+    ents <-
+        -- filterM (const (pure True)) -- isNotPersistEntityInstanceAlready
+        filterM shouldGenerateCode
+        $ embedEntityDefs
+        $ map (setDefaultIdFields mps)
+        $ ents'
```

Comment:
this is the Meat of the change

inlined the changes and did `filterM` to skip things already defined.